modules/nf-core/fastqc.nf

```diff
@@ -1,19 +1,31 @@
 process FASTQC {
-    label 'FASTQC'
+    tag "${meta.id}"
+    label 'process_medium'
     label 'cpus_2'
 
-    tag "${idPatient}-${idRun}"
+    publishDir "${params.outdir}/Reports/${meta.sample}/FastQC/${meta.id}",
+        mode: params.publish_dir_mode,
+        saveAs: { filename ->
+                    if (filename.endsWith('.version.txt')) null
+                    else filename }
 
-    publishDir "${params.outdir}/Reports/${idSample}/FastQC/${idSample}_${idRun}", mode: params.publish_dir_mode
+    container "quay.io/biocontainers/fastqc:0.11.9--0"
 
     input:
-        tuple val(idPatient), val(idSample), val(idRun), path("${idSample}_${idRun}_R1.fastq.gz"), path("${idSample}_${idRun}_R2.fastq.gz")
+        tuple val(meta), path(reads)
 
     output:
-        path "*.{html,zip}"
+        path "*.html",        emit: html
+        path "*.version.txt", emit: version
+        path "*.zip",         emit: zip
 
     script:
+    prefix = "${meta.id}"
     """
-    fastqc -t 2 -q ${idSample}_${idRun}_R1.fastq.gz ${idSample}_${idRun}_R2.fastq.gz
+    [ ! -f  ${prefix}_1.fastq.gz ] && ln -s ${reads[0]} ${prefix}_1.fastq.gz
+    [ ! -f  ${prefix}_2.fastq.gz ] && ln -s ${reads[1]} ${prefix}_2.fastq.gz
+    fastqc --threads ${task.cpus} ${prefix}_1.fastq.gz ${prefix}_2.fastq.gz
```

Comment:
Out of curiosity: What does [! -f ...] do?

Reply:
No idea, just copied it over from chipseq.
I figured it would be easier to update from the nf-core modules from there

Reply:
maybe @drpatelh then knows 😃

Reply:
Its bash notation for checking if a _file_ doesnt exist 🙂